### PR TITLE
🐛 fix order of docker stages

### PR DESCRIPTION
**Pull request by prashantgupta24**
Essentially the `release_test` should happen after release to control the order

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -56,7 +56,7 @@ RUN ./scripts/publish.sh
 # This phase installs the indicated version from PyPi and runs the unit tests
 # against the installed version.
 ##
-FROM base as release_test
+FROM release as release_test
```
Comment (prashantgupta24) [inline Dockerfile:59]: We have `base`-> `test`-> `release`, so ideally the `release_test` could just depend on `release` to ensure `release` is finished before running itself

Comment (gabe-l-hart) [reply Dockerfile:59]: I think if we do this, we need to explicitly remove the local copy of the code from `release`. Something like `RUN /src`. The whole reason it _didn't_ depend on `release` before was that I wanted to ensure it worked in a "clean room" setting.

**Review by gabe-l-hart — CHANGES_REQUESTED** [review]
I think we still want to ensure that the `release_test` doesn't have a local copy of the code kicking around so that it's a proper test

File: Dockerfile (hunk 1 above)
Comment (prashantgupta24) [reply Dockerfile:59]: Ah that makes sense.

Comment (prashantgupta24) [reply Dockerfile]: @gabe-l-hart how does this approach look like?

**Review by gabe-l-hart — APPROVED** [review]
Clever! Works for me

**Pull request merged**